service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/PlacementSubmittedEventControllerTest.java

```diff
@@ -0,0 +1,138 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.Placement;
+import uk.gov.hmcts.reform.fpl.model.PlacementOrderAndNotices;
+import uk.gov.hmcts.reform.fpl.model.Representative;
+import uk.gov.hmcts.reform.fpl.model.Respondent;
+import uk.gov.hmcts.reform.fpl.model.RespondentParty;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static java.util.UUID.randomUUID;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.NOTICE_OF_PLACEMENT_ORDER_UPLOADED_TEMPLATE;
+import static uk.gov.hmcts.reform.fpl.enums.RepresentativeServingPreferences.DIGITAL_SERVICE;
+import static uk.gov.hmcts.reform.fpl.model.PlacementOrderAndNotices.PlacementOrderAndNoticesType.NOTICE_OF_PLACEMENT_ORDER;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(PlacementController.class)
+@OverrideAutoConfiguration(enabled = true)
+class PlacementSubmittedEventControllerTest extends AbstractControllerTest {
+
+    @MockBean
+    private NotificationClient notificationClient;
+
+    PlacementSubmittedEventControllerTest() {
+        super("placement");
+    }
+
+    @Test
+    void shouldSendEmailNotificationWhenNewOrder() throws NotificationClientException {
+        postSubmittedEvent(callbackRequestWithEmptyCaseDetailsBefore());
+
+        verify(notificationClient, times(1)).sendEmail(
+            eq(NOTICE_OF_PLACEMENT_ORDER_UPLOADED_TEMPLATE),
+            eq("local-authority@local-authority.com"),
+            eq(parameters()),
+            eq("1"));
+
+        verify(notificationClient, times(1)).sendEmail(
+            eq(NOTICE_OF_PLACEMENT_ORDER_UPLOADED_TEMPLATE),
+            eq("representative@example.com"),
+            eq(parameters()),
+            eq("1"));
+    }
+
+    @Test
+    void shouldNotSendEmailNotificationWhenNoChangesToOrder() throws NotificationClientException {
+        postSubmittedEvent(callbackRequestWithMatchingCaseDetailsBefore());
+
+        verify(notificationClient, times(0)).sendEmail(
```

Comment:
Interesting, I used `org.mockito.Mockito.never` to verify similar. Is times(0) better? Or too similar to merit a change.

Reply:
I think never is better

Reply:
changed to never